packages/block-editor/src/components/link-control/index.js

```diff
@@ -194,14 +208,36 @@ function LinkControl( {
 		// If it's potentially a URL search then concat on a URL search suggestion
 		// just for good measure. That way once the actual results run out we always
 		// have a URL option to fallback on.
-		return couldBeURL && ! args.isInitialSuggestions
-			? results[ 0 ].concat( results[ 1 ] )
-			: results[ 0 ];
+		results =
+			couldBeURL && ! args.isInitialSuggestions
+				? results[ 0 ].concat( results[ 1 ] )
+				: results[ 0 ];
+
+		// Here we append a faux suggestion to represent a "CREATE" option. This
+		// is detected in the rendering of the search results and handeled as a
+		// special case. This is currently necessary because the suggestions
+		// dropdown will only appear if there are valid suggestions and
+		// therefore unless the create option is a suggestion it will not
+		// display in scenarios where there are no results returned from the
+		// API. In addition promoting CREATE to a first class suggestion affords
+		// the a11y benefits afforded by `URLInput` to all suggestions (eg:
+		// keyboard handling, ARIA roles...etc).
+		// Note also that due to the flow of data and the nature of the
+		// controlled components the value of the `url` property must correspond
+		// to the text value of the `<input>` otherwise it will result in an
+		// incorrectly named entity being created.
```

Comment:
I'm having a hard time following in the logic why we would expect that without the `url` property, an incorrectly-named entity would be created. We're not even passing the `url` through to the `createEntity` prop.

Reply:
I think this is due to a rebase changing `url` to `title`. I could have sworn it was using `url` before.

Anyway, the logic flows like this:

* User types in a value that we "Create".
* We show a "Create" suggestion which uses the current `<input />` value as it's `title`.
* User clicks on "Create" suggestion.
* `title` prop of the `suggestion` is passed to `createEntity` to create the new entity.

Sorry for the confusion.